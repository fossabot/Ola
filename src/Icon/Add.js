import React from 'react'

const Add = () => <path
  d="M14,2 C7.38326923,2 2,7.38326923 2,14 C2,20.6167308 7.38326923,26 14,26 C20.6167308,26 26,20.6167308 26,14 C26,7.38326923 20.6167308,2 14,2 Z M20.0018344,15.0161123 L15.0608255,15.0161123 L15.0608255,19.9803097 L13.0231809,19.9803097 L13.0231809,15.0161123 L8.01169143,15.0161123 L8.01169143,12.9724987 L13.0231809,12.9724987 L13.0231809,8.02789951 L15.0608255,8.02789951 L15.0608255,12.9724987 L20.0018344,12.9724987 L20.0018344,15.0161123 Z" />

export default Add