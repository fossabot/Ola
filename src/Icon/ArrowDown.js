import React from 'react'

const ArrowDown = () => <path
    d="M14 26l10.706-9.292a1 1 0 00-1.318-1.5L15 22.5V3a1 1 0 00-1.993-.117L13 22.5l-8.292-7.206c-.36-.361-1.037-.406-1.408-.02s-.396.945-.091 1.338L14 26z" />

export default ArrowDown
