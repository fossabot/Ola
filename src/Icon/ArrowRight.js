import React from 'react'

const ArrowRight = () => <path
  d="M26 14L16.708 3.293a1 1 0 00-1.5 1.319L22.5 13H3a1 1 0 00-.117 1.993L22.5 15l-7.207 8.292c-.36.36-.405 1.036-.02 1.408.386.37.946.395 1.339.09L26 14z" />

export default ArrowRight