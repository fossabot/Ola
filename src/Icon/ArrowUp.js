import React from 'react'

const ArrowUp = () => <path
  d="M14 2l10.706 9.292a1 1 0 01-1.318 1.5L15 5.5V25a1 1 0 01-1.993.117L13 5.5l-8.292 7.206c-.36.361-1.037.406-1.408.02s-.396-.945-.091-1.338L14 2z" />

export default ArrowUp
