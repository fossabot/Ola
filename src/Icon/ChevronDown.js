import React from 'react'

const SvgChevronDown = props => (
  <svg width={28} height={28} {...props}>
    <path
      d="M14 15.586l5.293-5.293a1 1 0 0 1 1.414 1.414l-6 6a1 1 0 0 1-1.414 0l-6-6a1 1 0 0 1 1.414-1.414L14 15.586z"
    />
  </svg>
)

export default SvgChevronDown
