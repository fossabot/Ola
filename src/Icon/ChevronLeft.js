import React from 'react'

const SvgChevronLeft = props => (
  <svg width={28} height={28} {...props}>
    <path
      d="M12.414 14l5.293 5.293a1 1 0 0 1-1.414 1.414l-6-6a1 1 0 0 1 0-1.414l6-6a1 1 0 0 1 1.414 1.414L12.414 14z"
    />
  </svg>
)

export default SvgChevronLeft
