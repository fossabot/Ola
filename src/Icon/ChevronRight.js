import React from 'react'

const SvgChevronRight = props => (
  <svg width={28} height={28} {...props}>
    <path
      d="M15.586 14l-5.293-5.293a1 1 0 0 1 1.414-1.414l6 6a1 1 0 0 1 0 1.414l-6 6a1 1 0 0 1-1.414-1.414L15.586 14z"
    />
  </svg>
)

export default SvgChevronRight
