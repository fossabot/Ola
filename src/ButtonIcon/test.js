import React from 'react';
import { Icon } from '../Icon';
import { ButtonIcon } from './';
import renderer from 'react-test-renderer';

it('Close Button icon', () => {
    const tree = renderer
      .create(
        <ButtonIcon>
          <Icon name="close" />
        </ButtonIcon>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
});
