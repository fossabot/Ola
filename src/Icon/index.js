import React from 'react'
import {default as PT} from 'prop-types'
import cx from 'classnames'

// Icons
import ChevronDown from './ChevronDown'
import ChevronUp from './ChevronUp'
import ChevronLeft from './ChevronLeft'
import ChevronRight from './ChevronRight'
import Clock from './Clock'
import Close from './Close'
import Error from './Error'
import Help from './Help'
import Home from './Home'
import Page from './Page'
import Search from './Search'
import Success from './Success'
import User from './User'
import Warning from './Warning'

const ICONS = {
  'chevronDown': ChevronDown,
  'chevronUp': ChevronUp,
  'chevronLeft': ChevronLeft,
  'chevronRight': ChevronRight,
  'clock': Clock,
  'close': Close,
  'error': Error,
  'help': Help,
  'home': Home,
  'page': Page,
  'search': Search,
  'success': Success,
  'user': User,
  'warning': Warning
}

const Icon = ({ name, size, extraClass, ...props }) => {
  const SpecificIcon = ICONS[name]
  const IconClasses = cx('ola_icon', `is-${size}`, extraClass)
  return (
    <svg width="28" height="28" viewBox="0 0 28 28" fillRule="evenodd" className={IconClasses} {...props}>
      <SpecificIcon />
    </svg>
  )
}

Icon.defaultProps = {
  name: 'help',
  size: 'medium',
  extraClass: null
}

Icon.propTypes = {
  /** Icons */
  name: PT.oneOf([
    'chevronDown',
    'chevronUp',
    'chevronLeft',
    'chevronRight',
    'clock',
    'close',
    'error',
    'help',
    'home',
    'page',
    'search',
    'success',
    'user',
    'warning'
  ]),
  /** Size */
  size: PT.oneOf(['small', 'medium', 'big']),
  /** Extra className */
  extraClass: PT.string
}

export default Icon
