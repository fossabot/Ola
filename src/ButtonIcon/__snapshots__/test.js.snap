// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Close Button icon 1`] = `
<button
  className="ola_buttonIcon"
>
  <svg
    className="ola_icon is-medium"
    fillRule="evenodd"
    height="28"
    viewBox="0 0 28 28"
    width="28"
  >
    <path
      d="M22.92 4c.296 0 .55.105.762.314.212.21.318.462.318.758s-.106.55-.318.762l-8.17 8.174 8.17 8.166c.212.212.318.464.318.754 0 .296-.106.549-.318.758-.212.21-.466.314-.762.314s-.547-.103-.753-.31l-8.171-8.166-8.163 8.166c-.206.207-.46.31-.761.31a1.04 1.04 0 0 1-.758-.31A1.02 1.02 0 0 1 4 22.936c0-.301.103-.555.31-.762l8.17-8.166-8.17-8.174A1.026 1.026 0 0 1 4 5.08c0-.296.105-.55.314-.762.21-.212.462-.318.758-.318.295 0 .55.106.761.318l8.163 8.174 8.17-8.174A1.03 1.03 0 0 1 22.92 4z"
    />
  </svg>
</button>
`;
